### ISSUE TITLE
Angular5 compatibility with build optimizer

### ISSUE THREAD

**Issue by pegaltier** [bergben/ng2-img-max#15]
In order to be compatible with angular5 and angular cli v1.5.0-rc.8 (new build optimizer) some files of the ng2-img-max must be modified.

FILES:
ng2-img-max.service
img-max-size.service
img-maxpx-size.service

FIX:
```
// import { Observable } from 'rxjs'; // bug: build optimizer
// import { Subject } from 'rxjs'; // bug: build optimizer
// import { Subject } from 'rxjs/Subject'; // ng5 fix
// import { Observable } from 'rxjs/Observable'; // ng5 fix
// import { BehaviorSubject } from 'rxjs/BehaviorSubject'; // ng5 fix
```

PACKAGE.JSON
```
"dependencies": {
    "@angular/animations": "^5.0.0",
    "@angular/common": "^5.0.0",
    "@angular/compiler": "^5.0.0",
    "@angular/core": "^5.0.0",
    "@angular/forms": "^5.0.0",
    "@angular/http": "^5.0.0",
    "@angular/platform-browser": "^5.0.0",
    "@angular/platform-browser-dynamic": "^5.0.0",
    "@angular/router": "^5.0.0",
    "@ngui/auto-complete": "^0.16.0",
    "@ngx-translate/core": "^8.0.0",
    "@ngx-translate/http-loader": "2.0.0",
    "@ultimate/ngxerrors": "^1.3.0",
    "angular2-moment": "^1.7.0",
    "blueimp-canvas-to-blob": "^3.14.0",
    "core-js": "^2.5.1",
    "ng2-auto-complete": "^0.12.0",
    "ng2-img-max": "^2.1.6",
    "ng2-toastr": "^4.1.2",
    "ngx-bootstrap": "^v2.0.0-beta.7",
    "ngx-img-fallback": "^1.2.0",
    "ngx-infinite-scroll": "^0.6.1",
    "ngx-progressbar": "^2.1.1",
    "rxjs": "5.5.2",
    "ts-helpers": "^1.1.2",
    "zone.js": "^0.8.18"
  },
  "devDependencies": {
    "@angular/cli": "^v1.5.0-rc.8",
    "@angular/compiler-cli": "^5.0.0",
    "@types/jasmine": "2.6.2",
    "@types/node": "^8.0.47",
    "codelyzer": "~3.2.2",
    "jasmine-core": "2.8.0",
    "jasmine-spec-reporter": "4.2.1",
    "karma": "1.7.1",
    "karma-chrome-launcher": "^2.2.0",
    "karma-cli": "^1.0.1",
    "karma-jasmine": "^1.1.0",
    "karma-remap-istanbul": "^0.6.0",
    "protractor": "5.2.0",
    "ts-node": "3.3.0",
    "tslint": "^5.8.0",
    "typescript": "~2.4.2",
    "webdriver-manager": "12.0.6"
  }
```

**Comment by dgaletto** [comment]
I sent PR #16.
Please check @bergben 

Regards

**Comment by bergben** [comment]
Hey @pegaltier and thanks for reporting this issue and an even bigger thanks to @dgaletto  for the PR. I just published the changes in version 2.1.7
